src/strategies/FluentStrategy.ts

```diff
@@ -53,34 +66,45 @@ export class FluentStrategy implements FluentStrategyInterface {
   /**
    * Hook that acts as point of extension of the addArbitrary function and that enables the strategy to be cached.
    */
-  setArbitraryCache<K extends string, A>(_arbitraryName: K) {}
+  setArbitraryCache<K extends string>(_arbitraryName: K) {}
 
   /**
    * Generates a once a collection of inputs for a given arbitrary
    */
   buildArbitraryCollection<A>(arbitrary: Arbitrary<A>, sampleSize = this.configuration.sampleSize): FluentPick<A>[] {
-    return this.isDedupable() ? arbitrary.sampleUnique(sampleSize) : arbitrary.sample(sampleSize)
+    const constantsSample = this.getArbitraryExtractedConstants(arbitrary)
+    return this.isDedupable() ?
+      arbitrary.sampleUnique(sampleSize, constantsSample) :
+      arbitrary.sample(sampleSize, constantsSample)
   }
 
   /**
    * Hook that acts as point of extension of the configArbitrary function and that enables an arbitrary to be shrinked.
    */
   shrink<K extends string>(_name: K, _partial: FluentResult | undefined) {}
 
+  /**
+   * Hook that acts as point of extension of the buildArbitraryCollection function and that enables the strategy to use
+   * extracted constants from code in the test cases.
+   */
+  getArbitraryExtractedConstants<A>(_arbitrary: Arbitrary<A>): FluentPick<A>[] {
```

Comment:
Can this hook be used for other purposes than extracting constants?

Reply:
As of right now, it is only being used for that purpose. I think that in the future and according to our needs we can rename it to something else.